packages/webpack-cli/lib/groups/resolveConfig.js

```diff
@@ -176,8 +179,9 @@ const finalize = async (moduleObj, args) => {
 
     if (configName) {
         const foundConfigNames = [];
+        const configsToFilter = isFunctionalConfig ? configs[0] : configs;
```

Comment:
Can you clarify why we need logic here? `configs` already contains configurations, maybe I am missing something?

Reply:
It's needed when config returns an array of configs.

in that case, `configs` is -
```js
[
  [
    {
      output: [Object],
      name: 'first',
      entry: './src/first.js',
      mode: 'development',
      stats: 'minimal'
    },
    {
      output: [Object],
      name: 'second',
      entry: './src/second.js',
      mode: 'production',
      stats: 'minimal'
    }
  ]
]
```

Reply:
I think we need to update this logic as this happens only when an array of configs is returned.

Reply:
Weird, I think we should have `[config, config, config]` here

Reply:
hmm, I'm looking into it.